File: TravellerCreatorModels/Basic/BasicStarSystem.cs
using TravellerCreatorGlobalMethods;
using TravellerCreatorModels.Enums;
using TravellerCreatorModels.Interfaces;

namespace TravellerCreatorModels.Basic;

public class BasicStarSystem : IStarSystem
{
    public List<IStar> Stars { get; set; } = new();
    public StarSystemType Type { get; set; } = StarSystemType.Basic;
    public IPlanet? Planet { get; set; }
    public List<IPlanet> Planets { get; set; } = new();
    public bool GasGiant { get; set; }

    public void GetGasGiant()
    {
        if (Roll.D6(2) > 5) {
            GasGiant = true;
        } else if (Roll.D6(2) < 5) {
            GasGiant = true;
        }
    }
}

File: TravellerCreator/ClientApp/src/app/sector-creator/models/basic/basic-star-system.ts
import {IStarSystem} from "../interfaces/star-system";
import {StarSystemType} from "../enums/star-system-type";
import {BasicPlanet} from "./basic-planet";
import {IStar} from "../interfaces/star";

export class BasicStarSystem implements IStarSystem {
  type: StarSystemType;
  gasGiant: boolean;
  planet: BasicPlanet;
  planets: BasicPlanet[];
  stars: IStar[];
}
